Github action publish to pypy broken

Successfully built a new minor release ([v0.2.2](https://github.com/dlr-eoc/ukis-csmask/releases/tag/v0.2.2)) but [cannot publish it to pypy](https://github.com/dlr-eoc/ukis-csmask/actions/runs/11454925795/job/31871805581). Github action `build and publish` returns 
```
ERROR    HTTPError: 403 Forbidden from https://upload.pypi.org/legacy/          
         Invalid or non-existent authentication information. See                
         https://pypi.org/help/#invalid-auth for more information.              
Error: Process completed with exit code 1.
```
There seems to be an issue with the user credentials on pypy, but I do not have the permissions to modify secrets or user credentials. @kambrium can you please have a look at this?

--- a/.github/workflows/python-publish.yml
+++ b/.github/workflows/python-publish.yml
@@ -5,22 +5,12 @@ on:
     types: [published]
 
 jobs:
-  deploy:
+  pypi-publish:
+    name: upload release to PyPI
     runs-on: ubuntu-latest
+    environment: pypi
+    permissions:
+      id-token: write
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Python
-      uses: actions/setup-python@v4
-      with:
-        python-version: '3.x'
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install setuptools wheel twine
-    - name: Build and publish
-      env:
-        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
-        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-      run: |
-        python setup.py sdist bdist_wheel
-        twine upload dist/*
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,7 +1,7 @@
 Changelog
 =========
 
-[1.0.0] (2024-11-XX)
+[1.0.0] (2025-02-14)
 --------------------
 Added
 *******
@@ -13,6 +13,7 @@ Changed
 *******
 - retrained all models with new architecture and training data
 - removed backwards compatibility with old band naming scheme
+- updated pypi publisher
 
 Fixed
 *******

